llm/src/services/external_api/dify.py

```diff
@@ -0,0 +1,97 @@
+import json
+import logging
+import os
+from typing import Dict, List
+
+import httpx
+from dotenv import load_dotenv
+from google.protobuf.json_format import MessageToDict
+
+from llm.v1.llm_pb2 import TalkHistory
+
+# 環境変数の読み込み
+load_dotenv()
+
+logger = logging.getLogger(__name__)
+
+DIFY_API_URL = os.getenv("DIFY_API_URL")
+DIFY_API_KEY = os.getenv("DIFY_API_KEY")
+
+if not DIFY_API_URL or not DIFY_API_KEY:
+    raise ValueError(
+        "DIFY_API_URL and DIFY_API_KEY must be set in environment variables"
+    )
+
+
+class DifyClient:
+    """Dify APIクライアント"""
+
+    def __init__(self) -> None:
+        self.client = httpx.AsyncClient(
+            base_url=DIFY_API_URL,
+            headers={
+                "Authorization": f"Bearer {DIFY_API_KEY}",
+                "Content-Type": "application/json",
+            },
+            timeout=30.0,  # タイムアウトを30秒に設定
+        )
+
+    async def close(self) -> None:
+        """クライアントをクローズ"""
+        await self.client.aclose()
+
+    async def __aenter__(self) -> "DifyClient":
+        """非同期コンテキストマネージャーのエントリー"""
+        return self
+
+    async def __aexit__(self, exc_type, exc_val, exc_tb) -> None:
+        """非同期コンテキストマネージャーのイグジット"""
+        await self.close()
+
+
+# シングルトンクライアントのインスタンス
+dify_client = DifyClient()
+
+
+async def call_dify_talk(input_histories: List[TalkHistory]) -> Dict[str, List[str]]:
```

Comment:
Listとかpython3.9からtypingをしなくても書けるようになりましたので、その書き方推奨です
```suggestion
async def call_dify_talk(input_histories: list[TalkHistory]) -> dict[str, list[str]]:
```